systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2IsolatedST.java

```diff
@@ -351,115 +251,90 @@ void testMirrorMaker2TlsAndTlsClientAuth(ExtensionContext extensionContext) thro
+            .withUserName(kafkaUserTargetName)
             .build();
 
-        LOGGER.info("Consumer in target cluster and topic should receive {} messages", messagesCount);
-        internalKafkaClient.checkProducedAndConsumedMessages(
-            sent,
-            internalKafkaClient.receiveMessagesTls()
-        );
+        LOGGER.info("Now messages should be mirrored to target topic and cluster");
+        resourceManager.createResource(extensionContext, clients.consumerTlsStrimzi(kafkaClusterTargetName));
+        ClientUtils.waitForClientSuccess(testStorage.getConsumerName(), testStorage.getNamespaceName(), MESSAGE_COUNT);
         LOGGER.info("Messages successfully mirrored");
 
-        KafkaTopic mirroredTopic = KafkaTopicResource.kafkaTopicClient().inNamespace(namespaceName).withName(topicTargetName).get();
+        KafkaTopicUtils.waitForKafkaTopicCreation(testStorage.getNamespaceName(), sourceMirroredTopicName);
+        KafkaTopic mirroredTopic = KafkaTopicResource.kafkaTopicClient().inNamespace(testStorage.getNamespaceName()).withName(sourceMirroredTopicName).get();
+
         assertThat(mirroredTopic.getSpec().getPartitions(), is(3));
         assertThat(mirroredTopic.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL), is(kafkaClusterTargetName));
-
-        mirroredTopic = KafkaTopicResource.kafkaTopicClient().inNamespace(namespaceName).withName(topicSourceNameMirrored).get();
-        assertThat(mirroredTopic, nullValue());
     }
 
     /**
      * Test mirroring messages by MirrorMaker 2.0 over tls transport using scram-sha-512 auth
      */
-    @SuppressWarnings({"checkstyle:MethodLength"})
     @ParallelNamespaceTest
     void testMirrorMaker2TlsAndScramSha512Auth(ExtensionContext extensionContext) {
-        final String namespaceName = StUtils.getNamespaceBasedOnRbac(INFRA_NAMESPACE, extensionContext);
-        final String kafkaClientsName = mapWithKafkaClientNames.get(extensionContext.getDisplayName());
-        String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
-        String kafkaClusterSourceName = clusterName + "-source";
-        String kafkaClusterTargetName = clusterName + "-target";
-        String sourceTopicName = "availability-topic-source-" + mapWithTestTopics.get(extensionContext.getDisplayName());
-        String targetTopicName = "availability-topic-target-" + mapWithTestTopics.get(extensionContext.getDisplayName());
-        String topicSourceNameMirrored = kafkaClusterSourceName + "." + sourceTopicName;
-        String topicSourceName = MIRRORMAKER2_TOPIC_NAME + "-" + rng.nextInt(Integer.MAX_VALUE);
-        String topicTargetName = kafkaClusterSourceName + "." + topicSourceName;
-        String kafkaUserSource = clusterName + "-my-user-source";
-        String kafkaUserTarget = clusterName + "-my-user-target";
+        final TestStorage testStorage = new TestStorage(extensionContext, INFRA_NAMESPACE);
```

Comment:
It would be better to change it to
```suggestion
        final TestStorage testStorage = new TestStorage(extensionContext, clusterOperator.getDeploymentNamespace);
```